### DISCUSSION TITLE
how to use wss in fastAPI

### DISCUSSION THREAD

**Discussion by Hui-cd** [fastapi/fastapi#6077] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.websocket("/wss")
```

### Description

how to use wss 

### Operating System

macOS

### FastAPI Version

0.88.0

### Python Version

3.11

**Comment by r0b2g1t** [comment, marked as answer]
@Hui-cd Do you tried the WebSockets implementation example from the docs in combination with the uvicorn running with https setup?

[https://fastapi.tiangolo.com/advanced/websockets/](https://fastapi.tiangolo.com/advanced/websockets/)

[https://www.uvicorn.org/deployment/#running-with-https](https://www.uvicorn.org/deployment/#running-with-https)

**Comment by arkadybag** [comment]
@Hui-cd Hello!
Unfortunately, need more details about your architecture. Is your application deployed behind web-server(nginx, etc...)?
If yes, you can terminate ssl connection on the web-server.

Also, a secure websocket connection starts with `wss://`, that is not a part of the url that you should configure on the app level

**Comment by caeser1996** [comment]
using websockets:
```
from fastapi import FastAPI
import websockets

app = FastAPI()

@app.get("/")
async def get():
    return {"message": "Hello, WebSockets"}

@app.websocket("/ws")
async def websocket_endpoint(websocket, path):
    await websocket.send("Hello, client!")
    while True:
        data = await websocket.recv()
        await websocket.send(f"You said: {data}")

```

**Reply by tiangolo** [reply]
Hey there, it seems you are copying output from ChatGPT or similar, please don't do that, the content generated by those systems is worse than no answer at all, as it not only tends to lie, but also does it in a very convincing way, and that causes more damage than benefit.

FastAPI does have builtin support for WebSockets: https://fastapi.tiangolo.com/advanced/websockets/
